File: 프로그래머스/unrated/132265. 롤케이크 자르기/롤케이크 자르기.js
function solution(topping) {
    const [lDic, lSizeArr, rDic, rSizeArr] = [new Set(), [], new Set(), []];

    topping.forEach((el, idx) => {
        lDic.add(el);

        if (idx !== 0) rDic.add(topping.at(-idx));

        lSizeArr.push(lDic.size);
        rSizeArr.push(rDic.size);
    });
    return lSizeArr.reduce((acc, el, idx) => (el === rSizeArr.at(-idx - 1)) ? ++acc : acc, 0);
}

File: 프로그래머스/unrated/132265. 롤케이크 자르기/롤케이크 자르기.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>
#include <numeric>

using namespace std;

int solution(vector<int> topping) {
    map<int, int> a, b;
    
    std::for_each(topping.begin(), topping.end(), [&](const auto& t) { ++b[t]; });
    return std::accumulate(topping.begin(), topping.end(), 0, [&](auto acc, const auto& t) {
        ++a[t], --b[t];
        if (b[t] == 0) b.erase(t);
        return (a.size() == b.size()) ? acc + 1 : acc;
    });
}
